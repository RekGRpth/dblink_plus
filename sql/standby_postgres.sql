-- dblink.query() to test use_xa GUC parameter on Standby Server
BEGIN;
SET dblink_plus.use_xa TO 'on';
SHOW dblink_plus.use_xa;
SELECT * FROM dblink.query('server_postgres6789012345678901234567890123456789012345678901234567890', 'SELECT * FROM dblink.dblink_tbl ORDER BY id') AS t(id integer, value text) ORDER BY id;
COMMIT;
BEGIN;
SET dblink_plus.use_xa TO 'off';
SELECT * FROM dblink.query('server_postgres6789012345678901234567890123456789012345678901234567890', 'SELECT * FROM dblink.dblink_tbl ORDER BY id') AS t(id integer, value text) ORDER BY id;
COMMIT;
