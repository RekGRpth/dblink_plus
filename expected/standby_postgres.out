-- dblink.query() to test use_xa GUC parameter on Standby Server
BEGIN;
SET dblink_plus.use_xa TO 'on';
SHOW dblink_plus.use_xa;
 dblink_plus.use_xa 
--------------------
 on
(1 row)

SELECT * FROM dblink.query('server_postgres6789012345678901234567890123456789012345678901234567890', 'SELECT * FROM dblink.dblink_tbl ORDER BY id') AS t(id integer, value text) ORDER BY id;
NOTICE:  identifier "server_postgres6789012345678901234567890123456789012345678901234567890" will be truncated to "server_postgres678901234567890123456789012345678901234567890123"
ERROR:  cannot execute DELETE in a read-only transaction
CONTEXT:  SQL statement "DELETE FROM dblink.atcommit"
COMMIT;
BEGIN;
SET dblink_plus.use_xa TO 'off';
SELECT * FROM dblink.query('server_postgres6789012345678901234567890123456789012345678901234567890', 'SELECT * FROM dblink.dblink_tbl ORDER BY id') AS t(id integer, value text) ORDER BY id;
NOTICE:  identifier "server_postgres6789012345678901234567890123456789012345678901234567890" will be truncated to "server_postgres678901234567890123456789012345678901234567890123"
 id | value 
----+-------
  1 | A
  2 | BB
  3 | CCC
  4 | DDDD
  5 | 
(5 rows)

COMMIT;
